### PR TITLE
CryptoSDK node.js C++ addon (using the latest N-API)

### PR DESCRIPTION
**Pull request by lbeder**

### CODE REVIEW

**Comment by lbeder** [comment]
@idandash @netoneko other than the integration to the build system - this PR is finished

File: client/crypto-sdk-javascript/src/address.cpp [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    napi_value jsthis;
+    status = napi_get_cb_info(env, info, nullptr, nullptr, &jsthis, nullptr);
+    assert(status == napi_ok);
+
+    Address* obj;
+    status = napi_unwrap(env, jsthis, reinterpret_cast<void**>(&obj));
+    assert(status == napi_ok);
+
+    napi_value version;
+    status = napi_create_double(env, obj->address_.GetVersion(), &version);
```
[the remark this answers is not in the document]
Comment (lbeder) [reply client/crypto-sdk-javascript/src/address.cpp]: Is bug 😅

File: e2e/test-from-docker.sh [hunk 2]
```diff
@@ -13,4 +13,4 @@ docker run -ti --rm --privileged  \
 -v /var/run/docker.sock:/var/run/docker.sock \
 -v $ROOT_DIR/logs:/opt/orbs/logs \
 $DOCKER_IMAGE:$DOCKER_TAG  \
-bash -c 'apt-get update && apt-get install -y python-pip python-dev && pip install docker-compose && (curl -fsSL get.docker.com | bash) && cd e2e && ./build.sh && yarn test'
```
Comment (netoneko) [inline e2e/test-from-docker.sh]: I remember I needed that to install docker-compose, but if it works without it, great!

File: client/crypto-sdk-javascript/test/address.spec.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    expect(address.version).to.equal(0);
+    expect(address.accountId).to.equal("c13052d8208230a58ab363708c08e78f1125f488");
+    expect(address.checksum).to.equal("50f8658b");
+    expect(address.toString()).to.equal("1JcVJcBXwqeVcC8T2nTpG2dT6xGzhdrHai");
+
+    const publicKey2 = "7a463487bb0eb584dabccd52398506b4a2dd432503cc6b7b582f87832ad104e6";
+    const address2 = new Address(publicKey2);
+
+    expect(address2.version).to.equal(0);
+    expect(address2.accountId).to.equal("44068acc1b9ffc072694b684fc11ff229aff0b28");
```
Comment (netoneko) [inline client/crypto-sdk-javascript/test/address.spec.ts]: giphy

**Pull request merged**